libgpos/src/_api.cpp

```diff
@@ -219,12 +219,12 @@ int gpos_exec
 			<< std::endl;
 
 		// unexpected failure
-		return 1;
+		throw ex;
```

Comment:
Since we change the behavior of this function, we also need to update comments saying: if success, return 0, else, return 1. If hit an exception, propagate the exception. For CAutoTaskProxy, make the `true` as default to propagate Error (exception).